src/Microsoft.AspNetCore.Razor.Evolution/RazorProjectItem.cs

```diff
@@ -34,6 +34,11 @@ public abstract class RazorProjectItem
         public abstract Stream Read();
 
         /// <summary>
+        /// Gets a value that determines if the project item exists.
+        /// </summary>
+        public abstract bool Exists { get; }
```

Comment:
We `Watch` files that do not exist so the cache can be invalidated if they appear later. This would allow using this API in the cases where we want all the paths